flake.nix

```diff
@@ -155,6 +155,8 @@
               # support having different crate types depending on the target, so
               # we switch there
               sed -i 's/\[lib\]/[lib]\ncrate-type = ["cdylib", "rlib"]/' Cargo.toml
+
+              sed -i '/utilities/d' Cargo.toml
```

Comment:
Is this really required? Patching `Cargo.toml` is a necessary evil but I would keep it to the minimum. If it doesn't compile to WASM because it uses io or `PathBuf` or whatever, can't we achieve the same with pure cargo, by excluding utilities when the feature `repl-wasm` (or something like that) is enabled?

Reply:
Sadly not as far as I could figure out.
Putting it behind a feature, or target did not prevent fs2 from being built.
I even tried putting every use of criterion behind another feature in the source.

There may be an incantation that works that I did not try, but I spent a silly amount of time trying to do this right, and failing. Hopefully this gets fixed on criterion end, there's already a issue for wasm. Otherwise eventually bench only deps may make it to cargo.

Reply:
I see. Then maybe you could document it quickly here (with even a link to the issue).

Reply:
Added